### PR TITLE
bpo-35733: `isinstance(ast.Constant(True), ast.Num)` is `False`

### PR DESCRIPTION
**Pull request by asottile**
https://bugs.python.org/issue35733

### CODE REVIEW

File: Lib/ast.py [hunk 1]
```diff
@@ -346,7 +346,10 @@ def __instancecheck__(cls, inst):
             except AttributeError:
                 return False
             else:
-                return isinstance(value, _const_types[cls])
+                return (
+                    isinstance(value, _const_types[cls]) and
+                    not isinstance(value, _const_types_not[cls])
```
Comment (serhiy-storchaka) [inline Lib/ast.py]:
```suggestion
                    not isinstance(value, _const_types_not.get(cls, ()))
```
and you can remove mappings to `()` in `_const_types_not`.

File: Lib/test/test_ast.py [hunk 2]
```diff
@@ -406,6 +406,8 @@ def test_isinstance(self):
         self.assertTrue(isinstance(ast.Constant(None), ast.NameConstant))
         self.assertTrue(isinstance(ast.Constant(...), ast.Ellipsis))
 
+        self.assertFalse(isinstance(ast.Constant(True), ast.Num))
```
Comment (serhiy-storchaka) [inline Lib/test/test_ast.py]: Please move these tests to the next group, after the test for `isinstance(ast.Constant('42'), ast.Num)`. Instead add the following test in this group after the test for `isinstance(ast.Str('42'), ast.Num)`:
```python
self.assertFalse(isinstance(ast.NameConstant(True), ast.Num))
```

File: Misc/NEWS.d/next/Library/2019-01-13-18-42-41.bpo-35733.eFfLiv.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+``isinstance(ast.Constant(True), ast.Num)`` is ``False``.  Patch by Anthony
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2019-01-13-18-42-41.bpo-35733.eFfLiv.rst]:
```suggestion
``ast.Constant(boolean)`` no longer an instance of :class:`ast.Num`.  Patch by Anthony
```

File: Misc/NEWS.d/next/Library/2019-01-13-18-42-41.bpo-35733.eFfLiv.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
...
+Sottile
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2019-01-13-18-42-41.bpo-35733.eFfLiv.rst]:
```suggestion
Sottile.
```

**Review by serhiy-storchaka — COMMENTED** [review]
In general LGTM.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by asottile** [comment]
@serhiy-storchaka anything left for me to do here?

**Pull request merged**